heartbeat/pgsql

```diff
@@ -1977,6 +1979,43 @@ check_log_file() {
     return 0
 }
 
+#
+# Check if we need to create stats temp directory in tmpfs
+#
+
+check_stat_temp_directory() {
+    local stats_temp
+
+    stats_temp=`get_pgsql_param stats_temp_directory`
+
+    if [ -z "$stats_temp" ]; then
+        return
+    fi
+
+    if [ "${stats_temp#/}" = "$stats_temp" ]; then
+        stats_temp="$OCF_RESKEY_pgdata/$stats_temp"
+    fi
+
+    if [ -d "$stats_temp" ]; then
+        return
+    fi
+
+    if ! mkdir -p "$stats_temp"; then
+        ocf_exit_reason "Can't create directory $stats_temp"
+        exit $OCF_ERR_PERM
+    fi
+
+    if ! chown $OCF_RESKEY_pgdba: "$stats_temp"; then
+        ocf_exit_reason "Can't change ownership for $stats_temp"
+        exit $OCF_ERR_PERM
+    fi
+
+    if ! chmod 750 "$stats_temp"; then
```

Comment:
Is there a reason that the permission is 750? All other directories in PGDATA have 700 permission by defaults so 700 seems like more appropriate for me.

Reply:
750 is the perms used by the Debian startup scripts when creating this directory but it seems to work with 700 too so I'll update the patch now.

Reply:
I see. I've only checked on RHEL/CentOS so it seems distributions dependent.
Thank you for making sure that it always works with 700 and updating the patch!